razorpay-subscriptions.php

```diff
@@ -40,39 +38,64 @@ function woocommerce_razorpay_subscriptions_init()
 
     class WC_Razorpay_Subscription extends WC_Razorpay
     {
+        /**
+         * Unique ID for the gateway
+         * @var string
+         */
         public $id = 'razorpay_subscriptions';
-        public $method_title = 'Razorpay Subscriptions';
 
-        const RAZORPAY_SUBSCRIPTION_ID       = 'razorpay_subscription_id';
-        const DEFAULT_LABEL                  = 'MasterCard/Visa Credit Card';
-        const DEFAULT_DESCRIPTION            = 'Setup automatic recurring billing on a MasterCard or Visa Credit Card';
+        /**
+         * Title of the payment method shown on the admin page.
+         * @var string
+         */
+        public $method_title = 'Razorpay Subscriptions';
 
+        /**
+         * This array controls what settings are visible to the user
+         * @var array
+         */
         protected $visibleSettings = array(
             'enabled',
             'title',
             'description',
         );
 
+        /**
+         * Contains all supported methods of woocommerce subscriptions
+         * @var array
+         */
         public $supports = array(
             'subscriptions',
             'subscription_reactivation',
             'subscription_suspension',
             'subscription_cancellation',
         );
 
+        /**
+         * Instance of the class RZP_subscriptions found in __DIR__ . 'includes/razorpay-subscriptions'
+         * It is used to make all subscriptions related API calls to the Razorpay's API
+         * @var RZP_Subscriptions
+         */
+        protected $subscriptions;
+
+        const RAZORPAY_SUBSCRIPTION_ID       = 'razorpay_subscription_id';
+        const DEFAULT_LABEL                  = 'MasterCard/Visa Credit Card';
+        const DEFAULT_DESCRIPTION            = 'Setup automatic recurring billing on a MasterCard or Visa Credit Card';
+
         public function __construct()
         {
             parent::__construct();
-            $this->icon =  plugins_url('images/logo.png' , __FILE__);
```

Comment:
This is needed, otherwise the parent one gets called, which will set its own logo